libraries/shared/logs/extractor.go

```diff
@@ -56,26 +72,42 @@ func (extractor *LogExtractor) AddTransformerConfig(config transformer.EventTran
 		return checkedHeadersErr
 	}
 
-	if extractor.StartingBlock == nil {
-		extractor.StartingBlock = &config.StartingBlockNumber
-	} else if earlierStartingBlockNumber(config.StartingBlockNumber, *extractor.StartingBlock) {
+	if resetStartingBlockNumber(config.StartingBlockNumber, extractor.StartingBlock) {
 		extractor.StartingBlock = &config.StartingBlockNumber
 	}
 
+	if resetEndingBlockNumber(config.EndingBlockNumber, extractor.EndingBlock) {
+		extractor.EndingBlock = &config.EndingBlockNumber
+	}
+
 	addresses := transformer.HexStringsToAddresses(config.ContractAddresses)
 	extractor.Addresses = append(extractor.Addresses, addresses...)
 	extractor.Topics = append(extractor.Topics, common.HexToHash(config.Topic))
 	return nil
 }
 
+func resetStartingBlockNumber(currentTransformerBlock int64, extractorBlock *int64) bool {
+	return extractorBlock == nil || currentTransformerBlock < *extractorBlock
+}
+
+func resetEndingBlockNumber(currentTransformerBlock int64, extractorBlock *int64) bool {
+	if extractorBlock == nil {
+		return true
+	} else if currentTransformerBlock == int64(-1) {
+		return true
+	}
+
+	return *extractorBlock != int64(-1) && currentTransformerBlock > *extractorBlock
```

Comment:
this is just a minor aesthetic thing (and maybe I've been looking at too much `dss` code 😂), but I'd be tempted to give these predicates names and combine them - something like:
```
isCurrentEndingBlockNil := extractorBlock == nil
isTransformerEndingBlockNil := currentTransformerBlock == int64(-1)
isCurrentEndingBlockAssigned := *extractorBlock != int64(-1)
isTransformerEndingBlockGreater := currentTransformerBlock > *extractorBlock

return isCurrentEndingBlockNil || isTransformerEndingBlockNil || (isCurrentEndingBlockAssigned && isTransformerEndingBlockGreater)
```

Reply:
Sounds good! The one thing is that we have to return from the method before setting `isCurrentEndingBlockAssigned` and `isTransformerEndingBlockGreater` if either of the blocks it's checking are nil. Otherwise we end up with a nil point exception.